tutorials/JsonWriterDefaultStream/JsonWriterDefaultStream.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigquerystorage;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.BigQueryOptions;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.Schema;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.bigquery.StandardTableDefinition;
+import com.google.cloud.bigquery.Table;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.TableInfo;
+import com.google.cloud.bigquery.storage.v1.AppendRowsResponse;
+import com.google.cloud.bigquery.storage.v1.JsonStreamWriter;
+import com.google.cloud.bigquery.storage.v1.TableName;
+import com.google.cloud.bigquery.storage.v1.TableSchema;
+import com.google.protobuf.Descriptors.DescriptorValidationException;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.concurrent.ExecutionException;
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+public class JsonWriterDefaultStream {
+
+  public static void main(String[] args) throws Exception {
+    if (args.length < 4) {
+      System.out.println("Arguments: project, dataset, table, source_file");
+      return;
+    }
+
+    String projectId = args[0];
+    String datasetName = args[1];
+    String tableName = args[2];
+    String dataFile = args[3];
+    createDestinationTable(projectId, datasetName, tableName);
+    writeToDefaultStream(projectId, datasetName, tableName, dataFile);
+  }
+
+  // createDestinationTable: Creates the destination table for streaming with the desired schema.
+  public static void createDestinationTable(
+      String projectId, String datasetName, String tableName) {
+    BigQuery bigquery = BigQueryOptions.getDefaultInstance().getService();
+    // Create a schema that matches the source data.
+    Schema schema =
+        Schema.of(
+            Field.of("commit", StandardSQLTypeName.STRING),
+            Field.newBuilder("parent", StandardSQLTypeName.STRING)
+                .setMode(Field.Mode.REPEATED)
+                .build(),
+            Field.of("author", StandardSQLTypeName.STRING),
```

Comment:
I think this is a good idea - Veronica can make some improvements after this is merged in.

Reply:
We could but I'm not sure it adds much to the sample. Creating the table isn't main purpose of the sample, and in the main body I'm just reading in JSON records and passing them to the writer.

Reply:
Looking at the schema again, I could change `time_sec` to a DateTime type, that actually makes the tutorial a little simpler, because my example query on the table currently uses `TIMESTAMP_SECONDS` to convert the value.